### PR TITLE
Fix marker popup console error

### PR DESCRIPTION
**Pull request by HarelM**
Fixes #211 as this issue had two bug associated with it.
This is done by removing the popup using the map service (mainly unregistering the events) even in the case where marker was already destroyed.

### CODE REVIEW

File: projects/ngx-maplibre-gl/src/lib/popup/popup.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      'addMarker',
+      'removeMarker',
+      'removePopupFromMap',
+      'createPopup',
+      'addPopupToMarker'
+    ],
+    {
+      mapCreated$: of(true),
+    }
+  );
```
Comment (ShellyDCMS) [inline projects/ngx-maplibre-gl/src/lib/popup/popup.component.spec.ts:20]: just wandering, can you use ts-stubber with jasmine.createSpy?

File: projects/ngx-maplibre-gl/src/lib/popup/popup.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  `,
+  standalone: true,
+  imports: [MarkerComponent, PopupComponent]
+})
+class MarkerPopupTestComponent {
+    show = true;
+    public toggle() {
+        this.show = !this.show;
+    }
```
Comment (ShellyDCMS) [inline projects/ngx-maplibre-gl/src/lib/popup/popup.component.spec.ts:40]: I don't see this method being called anywhere in the test - is this the API? a marker component is shown/hidden using a public toggle method?

Comment (HarelM) [reply projects/ngx-maplibre-gl/src/lib/popup/popup.component.spec.ts:40]: Right, I ended up using the property. I'll remove this.

File: projects/ngx-maplibre-gl/src/lib/popup/popup.component.spec.ts (hunk 1 above)
Comment (HarelM) [reply projects/ngx-maplibre-gl/src/lib/popup/popup.component.spec.ts:20]: What works be the benefit in this case? The downside is obviously another dependency to maintain...

**Review by marcjulian — APPROVED** [review]

**Pull request merged**